oktapam/resource_okta_universal_directory_checkout_settings_test.go

```diff
@@ -0,0 +1,342 @@
+package oktapam
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"sync"
+	"testing"
+
+	"github.com/atko-pam/pam-sdk-go/client/pam"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/jarcoal/httpmock"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+)
+
+var _ plancheck.PlanCheck = debugPlan{}
+
+type debugPlan struct{}
+
+func (e debugPlan) CheckPlan(ctx context.Context, req plancheck.CheckPlanRequest, resp *plancheck.CheckPlanResponse) {
+	rd, err := json.Marshal(req.Plan)
+	if err != nil {
+		fmt.Println("error marshalling machine-readable plan output:", err)
+	}
+	fmt.Printf("req.Plan - %s\n", string(rd))
+}
+
+func DebugPlan() plancheck.PlanCheck {
+	return debugPlan{}
+}
+
+const testAccOktaUDCheckoutSettingsBaseConfigFormat = `
+resource "oktapam_group" "test_resource_group_dga_group" {
+	name = "%s"
+}
+resource "oktapam_resource_group" "test_acc_resource_group" {
+	name = "%s"
+	description = "test resource group"
+	delegated_resource_admin_groups = [oktapam_group.test_resource_group_dga_group.id]
+}
+resource "oktapam_resource_group_project" "test_acc_resource_group_project" {
+	name = "%s"
+	resource_group = oktapam_resource_group.test_acc_resource_group.id
+	ssh_certificate_type  = "CERT_TYPE_ED25519_01"
+	account_discovery     = true
+}
+`
+
+const testAccOktaUDCheckoutSettingsCreateConfigFormat = `
+resource "oktapam_okta_universal_directory_checkout_settings" "test_acc_okta_universal_directory_checkout_settings" {
+	resource_group = oktapam_resource_group.test_acc_resource_group.id
+	project = oktapam_resource_group_project.test_acc_resource_group_project.id
+	checkout_required = true
+	checkout_duration_in_seconds = 900
+}
+`
+
+const testAccOktaUDCheckoutSettingsUpdateWithIncludeListConfigFormat = `
+resource "oktapam_okta_universal_directory_checkout_settings" "test_acc_okta_universal_directory_checkout_settings" {
+	resource_group = oktapam_resource_group.test_acc_resource_group.id
+	project = oktapam_resource_group_project.test_acc_resource_group_project.id
+	checkout_required = true
+	checkout_duration_in_seconds = 3600
+	include_list = [
```

Comment:
@shrirangadgaonkar-okta  does this make sense? Or should there be a resource/data source for okta universal directory resources?